src/Sentry/Protocol/Transaction.cs

```diff
@@ -0,0 +1,329 @@
+        public string Operation
+        {
+            get => Contexts.Trace.Operation;
+            internal set => Contexts.Trace.Operation = value;
+        }
+
+        /// <inheritdoc />
+        public string? Description { get; set; }
+
+        /// <inheritdoc />
+        public SpanStatus? Status
+        {
+            get => Contexts.Trace.Status;
+            private set => Contexts.Trace.Status = value;
+        }
+
+        /// <inheritdoc />
+        public bool IsSampled { get; set; }
+
+        /// <inheritdoc />
+        public SentryLevel? Level { get; set; }
+
+        private Request? _request;
+
+        /// <inheritdoc />
+        public Request Request
+        {
+            get => _request ??= new Request();
+            set => _request = value;
+        }
+
+        private Contexts? _contexts;
+
+        /// <inheritdoc />
+        public Contexts Contexts
+        {
+            get => _contexts ??= new Contexts();
+            set => _contexts = value;
+        }
+
+        private User? _user;
+
+        /// <inheritdoc />
+        public User User
+        {
+            get => _user ??= new User();
+            set => _user = value;
+        }
+
+        /// <inheritdoc />
+        public string? Environment { get; set; }
+
+        /// <inheritdoc />
+        public SdkVersion Sdk { get; internal set; } = new SdkVersion();
+
+        private IEnumerable<string>? _fingerprint;
+
+        /// <inheritdoc />
+        public IEnumerable<string> Fingerprint
+        {
+            get => _fingerprint ?? Enumerable.Empty<string>();
+            set => _fingerprint = value;
+        }
+
+        private List<Breadcrumb>? _breadcrumbs;
+
+        /// <inheritdoc />
+        public IEnumerable<Breadcrumb> Breadcrumbs => _breadcrumbs ??= new List<Breadcrumb>();
+
+        private Dictionary<string, object?>? _extra;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<string, object?> Extra => _extra ??= new Dictionary<string, object?>();
+
+        private Dictionary<string, string>? _tags;
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new Dictionary<string, string>();
+
+        private List<Span>? _children;
+
+        // Transaction never has a parent
+        SentryId? ISpanContext.ParentSpanId => null;
+
+        string? IScope.TransactionName
+        {
+            get => Name;
+            set => Name = value ?? "unnamed";
+        }
+
+        internal Transaction(IHub hub, IScopeOptions? scopeOptions)
+        {
+            _hub = hub;
+            ScopeOptions = scopeOptions;
+
+            SpanId = SentryId.Create();
+            TraceId = SentryId.Create();
+        }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="Transaction"/>.
+        /// </summary>
+        /// <summary>
+        /// Child spans.
+        /// </summary>
+        public IReadOnlyList<Span> Children => _children ??= new List<Span>();
```

Comment:
I think here we could go with ConcurrentBag and set the instance from the start, since we expect transactions to have spans

Reply:
I see that the fields are eagerly-initialized on `Span` (which is accessed concurrently) so maybe we should do this here too
![image](https://user-images.githubusercontent.com/1935960/101800165-a3990a80-3b15-11eb-8649-469ea3f1ef7b.png)